CMakeLists.txt

```diff
@@ -430,43 +477,83 @@ set(SOURCEFILES
     ${FEATURE_SOURCEFILES} 
     ${INTF_SOURCEFILES})
 
-if(${SHARED_LIB})
-    message(STATUS "Building shared library")
-    set(LIBTYPE SHARED)
-else()
-    message(STATUS "Building static library")
-    set(LIBYTPE STATIC)
-endif()
-
-add_library(pubnub ${LIBTYPE} ${SOURCEFILES})
-
-target_include_directories(pubnub PUBLIC
-    ${CMAKE_CURRENT_LIST_DIR}
-    ${CMAKE_CURRENT_LIST_DIR}/core 
-    ${CMAKE_CURRENT_LIST_DIR}/lib)
+if(NOT ESP_PLATFORM)
+    if(${SHARED_LIB})
+        message(STATUS "Building shared library")
+        set(LIBTYPE SHARED)
+    else()
+        message(STATUS "Building static library")
+        set(LIBYTPE STATIC)
+    endif()
 
-if(${WITH_CPP})
-    target_include_directories(pubnub PUBLIC ${CMAKE_CURRENT_LIST_DIR}/cpp)
-endif()
+    add_library(pubnub ${LIBTYPE} ${SOURCEFILES})
 
-if(${OPENSSL})
-    target_include_directories(pubnub PUBLIC 
-        ${CMAKE_CURRENT_LIST_DIR}/openssl/
-        ${CMAKE_CURRENT_LIST_DIR}/lib/base64)
-    target_include_directories(pubnub PUBLIC ${OPENSSL_INCLUDE_DIR})
+    target_include_directories(pubnub PUBLIC
+        ${CMAKE_CURRENT_LIST_DIR}
+        ${CMAKE_CURRENT_LIST_DIR}/core 
+        ${CMAKE_CURRENT_LIST_DIR}/lib)
+    
+    if(${WITH_CPP})
+        target_include_directories(pubnub PUBLIC ${CMAKE_CURRENT_LIST_DIR}/cpp)
+    endif()
+    
+    if(${OPENSSL})
+        target_include_directories(pubnub PUBLIC 
+            ${CMAKE_CURRENT_LIST_DIR}/openssl/
+            ${CMAKE_CURRENT_LIST_DIR}/lib/base64)
+        target_include_directories(pubnub PUBLIC ${OPENSSL_INCLUDE_DIR})
+    elseif(${MBEDTLS})
+        target_include_directories(pubnub PUBLIC ${CMAKE_CURRENT_LIST_DIR}/mbedtls)
+    else()
+        if(UNIX)
+            target_include_directories(pubnub PUBLIC ${CMAKE_CURRENT_LIST_DIR}/posix)
+        elseif(WIN32 OR WIN64 OR MSVC)
+            target_include_directories(pubnub PUBLIC
+                ${CMAKE_CURRENT_LIST_DIR}/windows
+                ${CMAKE_CURRENT_LIST_DIR}/core/c99)
+        else()
+            target_include_directories(pubnub PUBLIC ${CMAKE_CURRENT_LIST_DIR}/freertos)
+        endif()
+    endif()
+    
+    target_link_libraries(pubnub ${LDLIBS} ${SSL_LINK_LIBRARIES})
 else()
-    if(UNIX)
-        target_include_directories(pubnub PUBLIC ${CMAKE_CURRENT_LIST_DIR}/posix)
-    elseif(WIN32 OR WIN64 OR MSVC)
-        target_include_directories(pubnub PUBLIC
-            ${CMAKE_CURRENT_LIST_DIR}/windows
-            ${CMAKE_CURRENT_LIST_DIR}/core/c99)
+    message(STATUS "Building ESP32 IDF component")
+
+    set(EXTRA_CFLAGS ${EXTRA_CFLAGS} ${FLAGS})
+
+    if (${MBEDTLS})
+       idf_component_register(SRCS ${SOURCEFILES}
```

Comment:
When testing this branch with our existing codebase, the following files required by our project were not part of `SOURCEFILES` and thus not being compiled:

```cmake
${CMAKE_CURRENT_LIST_DIR}/mbedtls/pbpal_mbedtls.c
${CMAKE_CURRENT_LIST_DIR}/mbedtls/pbpal_connect_mbedtls.c
${CMAKE_CURRENT_LIST_DIR}/freertos/pbtimespec_elapsed_ms_esp32.c
```

Reply:
I played a little with CMake imports and those files also have been added.